### DISCUSSION TITLE
How do I support multiple refreshListenables?

### DISCUSSION THREAD

**Discussion by benPesso** [csells/go_router#189] [Q&A]
Pretty self explanatory, but it would make life easier.

Instead of having to do this:
```dart
refreshListenable: ref.read(routerRefreshProvider.notifier)
```

We could do this (_and maybe rename the property along the way to `refreshNotifiers`, since that's what it does_):
```dart
refreshNotifiers: [ref.read(authStatusProvider.notifier), ref.read(incomingAlertProvider.notifier), ...]
```

**Comment by csells** [comment]
I think it's pretty easy to create an aggregator that turns multiple listenables into a single listenable.

**Comment by benPesso** [comment]
It is, but it's not verbose enough, and it adds another degree of separation in the code. Creating an aggregator (as in my first example) is just needless overhead, in my opinion.

**Comment by letsar** [comment]
Hi @benPesso, you can use Listenable.merge for that I think.

**Comment by csells** [comment]
Pro tip! https://api.flutter.dev/flutter/foundation/Listenable/Listenable.merge.html
